cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskManager.java

```diff
@@ -87,21 +81,35 @@ public ExecutionTaskManager(int defaultInterBrokerPartitionMovementConcurrency,
 
   /**
    * Dynamically set the inter-broker partition movement concurrency per broker.
+   * Ensure that the requested concurrency is smaller than the maximum number of allowed partition movements in cluster.
    *
    * @param requestedInterBrokerPartitionMovementConcurrency The maximum number of concurrent inter-broker partition movements per broker
    *                                                         (if null, use {@link #_defaultInterBrokerPartitionMovementConcurrency}).
    */
   public synchronized void setRequestedInterBrokerPartitionMovementConcurrency(Integer requestedInterBrokerPartitionMovementConcurrency) {
+    if (requestedInterBrokerPartitionMovementConcurrency != null
+        && requestedInterBrokerPartitionMovementConcurrency >= _maxClusterPartitionMovementConcurrency) {
+      throw new IllegalArgumentException("Attempt to set inter-broker partition movement concurrency ["
+                                         + requestedInterBrokerPartitionMovementConcurrency + "] to greater than or equal to the maximum number"
+                                         + " of allowed partition movements in cluster [" + _maxClusterPartitionMovementConcurrency + "].");
+    }
     _requestedInterBrokerPartitionMovementConcurrency = requestedInterBrokerPartitionMovementConcurrency;
   }
 
   /**
    * Dynamically set the intra-broker partition movement concurrency.
+   * Ensure that the requested concurrency is smaller than the maximum number of allowed partition movements in cluster.
    *
    * @param requestedIntraBrokerPartitionMovementConcurrency The maximum number of concurrent intra-broker partition movements
    *                                                         (if null, use {@link #_defaultIntraBrokerPartitionMovementConcurrency}).
    */
   public synchronized void setRequestedIntraBrokerPartitionMovementConcurrency(Integer requestedIntraBrokerPartitionMovementConcurrency) {
+    if (requestedIntraBrokerPartitionMovementConcurrency != null
```

Comment:
we should also apply similar sanity check for requested leadership movement concurrency?